roles/openshift_efk/tasks/main.yaml

```diff
@@ -0,0 +1,179 @@
+---
+
+  - fail: msg="This role requires the following vars to be defined: master_url, kibana_hostname, es_cluster_size"
+    when: "kibana_hostname is not defined or
+          es_cluster_size is not defined or
+          master_url is not defined"
+
+  - name: "Checking for logging project"
+    command: oc get project logging
+    register: logging_project
+    failed_when: "'FAILED' in logging_project.stderr"
+    tags:
+      - cleanup
+
+  - name: "Create logging project"
+    command: oadm new-project logging
+    when: logging_project.rc != 0
+    tags:
+      - build
+  - name: "Changing projects"
+    command: oc project logging
```

Comment:
Yes, in general you will want to copy directly from the admin config in /etc/origin/master/ (there is an openshift.common variable that I can't recall off the top of my head that provides the path without hardcoding it), and use the copied version. You can still use the `-n` flag as needed, but the `-n` flag using the default kubeconfig does not guarantee that you will be able to perform the operation.

For example, a local user may have done an `oc login` using the default kubeconfig and then you would be attempting to operate as that other user.